charts/carbon-intensity-exporter/templates/deployment.yaml

```diff
@@ -24,10 +24,46 @@ spec:
         - name: {{ .Values.apiServer.name }}
           image: "{{ .Values.apiServer.image.repository }}:{{ .Values.apiServer.image.tag }}"
           imagePullPolicy: {{ .Values.apiServer.image.pullPolicy }}
-          command:
-            - sh
-            - -c
-            - cd /app && sed "s/username/{{ .Values.apiServer.username }}/" -i appsettings.json && sed "s/password/{{ .Values.apiServer.password }}/" -i appsettings.json && dotnet CarbonAware.WebApi.dll
+          env:
+          - name: DataSources__EmissionsDataSource
+            value: {{ .Values.providerName }}
+          - name: DataSources__ForecastDataSource
+            value: {{ .Values.providerName }}
+          {{- if eq .Values.providerName "ElectricityMaps" }}
+          - name: DataSources__Configurations__ElectricityMaps__Type
+            value: {{ .Values.providerName }}
+          - name: DataSources__Configurations__ElectricityMaps__BaseURL
+            value: {{ .Values.electricityMaps.baseURL }}
+          - name: DataSources__Configurations__ElectricityMaps__APITokenHeader
+            value: {{ .Values.electricityMaps.apiTokenHeader }}
+          - name: DataSources__Configurations__ElectricityMaps__APIToken
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "carbon-intensity-exporter.fullname" . }}
+                key: apiToken
```

Comment:
I'd like to get some feedback here. 

@helayoty My approach was to take the settings from the config file in your fork https://github.com/helayoty/carbon-aware-sdk/commit/55d75c78ba57429b01e2582f9f40aa4156a0d834 but to set them via env vars so the credentials can reference the secret.

We could also template the config file with helm but I'm not sure how to handle the secrets in that case.

If you'd prefer a different approach please let me know.

Reply:
This is great. I think for a containerized dotnet app, env vars as appsettings values is the best approach.